zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionAcknowledgeProcessor.java

```diff
@@ -57,7 +57,9 @@ public void processRecord(final TypedRecord<CommandDistributionRecord> record) {
     stateWriter.appendFollowUpEvent(
         distributionKey, CommandDistributionIntent.ACKNOWLEDGED, recordValue);
 
-    commandDistributionBehavior.distributeNextInQueue(distributionKey, partitionId);
+    distributionState
+        .getQueueIdForDistribution(distributionKey)
+        .ifPresent(queueId -> commandDistributionBehavior.advanceQueue(queueId, partitionId));
```

Comment:
❌ / 🔧  It feels a bit strange to me to try and continue in the Acknowledge processor. The acknowledgement comes from a single partition. Most of the times we'd still need to wait for the acknowledgement of the other partition before we can continue with the continuation command.

I find it cleaner to have a `CommandDistribution.FINISH` command that takes care of this, as well as writing the `FINISHED` event. What do you think?

Reply:
That would only shift things around slightly, no? Right now we check if the queue is done and if it is we write the continuation command. What you are proposing is to still check if the queue is done but then first write a `CommandDistribution.FINISH` command that then writes the continuation command.  What do we gain from that?
Perhaps a bit besides the point but I'd find `CommandDistribution.FINISH` confusing because it'd only happen when the queue becomes empty and `CommandDistribution.FINISHED` has an entirely different meaning and is written for each individual distribution.

Reply:
> What do we gain from that?

Semantically it makes sense to check for callbacks and trigger them after something has finished. So instead of the log being `Distribute -> Acknowledge -> Continuation Completed -> Finished`

It'd be `Distribute -> Acknowledge -> Finished -> Continuation Completed`
In my opinion the second option is more correct. After a distribution is finished you trigger the continuation, as the continuation is a final step.

You'd also be doing less useless checks. Within the acknowledge processor you know >50% of the processed commands aren't going to trigger a continuation, since other partitions haven't acknowledged yet. If you do it in a `FINISH` command you know all the partitions are done with the distribution and there's a fair chance it was the last distribution of the queue.

> Perhaps a bit besides the point but I'd find CommandDistribution.FINISH confusing because it'd only happen when the queue becomes empty and CommandDistribution.FINISHED has an entirely different meaning and is written for each individual distribution.

I could say the same for CommandDistribution.ACKNOWLEDGED 😄 This processor is even more atomic as it's written for each partial distribution.

Reply:
Exactly 👍

Reply:
Done, please check the latest commits :)